### ISSUE TITLE
"socket hang up" when using Proxy

### ISSUE THREAD

**Issue by Fndroid** [ddsol/speedtest.net#75]
I am using a http proxy with speedtest-net like below:
```
var speedTest = require('speedtest-net');

speedTest.visual({maxTime: 15000, proxy: "http://127.0.0.1:7890"}, (err, data) => {
  console.dir(err, data);
});

```

Then got an error:

```
{ Error: socket hang up
    at createHangUpError (_http_client.js:322:15)
    at Socket.socketOnEnd (_http_client.js:425:23)
    at Socket.emit (events.js:187:15)
    at endReadableNT (_stream_readable.js:1094:12)
    at process._tickCallback (internal/process/next_tick.js:63:19) code: 'ECONNRESET' }
```

**Comment by ddsol** [comment]
I don't know why your proxy closed the connection... I'll leave this open for a bit to see if anyone has input, but you should probably be able to debug this yourself. Check if you can make http request through your proxy. You can also set your browser to use it and see if that also breaks. Your proxy may not support https, which is required.

**Comment by Fndroid** [comment]
> I don't know why your proxy closed the connection... I'll leave this open for a bit to see if anyone has input, but you should probably be able to debug this yourself. Check if you can make http request through your proxy. You can also set your browser to use it and see if that also breaks. Your proxy may not support https, which is required.

I am actually using it as my system proxy. It works fine.

**Comment by ddsol** [comment]
I'll leave it open then, but it's probably faster to debug this yourself. If you come up with a cause and/or solution, post it here in the form of a PR or a comment so others can use it as well.

**Comment by Fndroid** [comment]
I tried another http proxy, and end up like https://github.com/ddsol/speedtest.net/issues/52 ..

**Comment by ddsol** [comment]
Probably be fixed in #82. If this is not the case, let me know and I will reopen.

**Comment by Fndroid** [comment]
works fine, thx
